lib/cisco_node_utils/bgp_af.rb

```diff
@@ -98,52 +98,34 @@ def set_args_keys(hash={}) # rubocop:disable Style/AccessorMethodName
     #                      PROPERTIES                      #
     ########################################################
 
-    # IOS XR?
-    def ios_xr?
-      platform == :ios_xr
+    def fail_unsupported
+      fail Cisco::UnsupportedError.new('','','')
     end
 
-    # Nexus?
-    def nexus?
-      platform == :nexus
-    end
+    # def vrf_unsupported_xr?
+    #   ios_xr? && @vrf != 'default'
+    # end
 
-    def vrf_unsupported_xr?
-      ios_xr? && @vrf != 'default'
-    end
-
-    def fail_if_vrf_unsupported_xr(reason)
-      fail Cisco::UnsupportedError.new(
-        'bgp_af', reason, 'set',
-        "#{reason} is not configurable on a " \
-        'per-VRF basis on IOS XR') if vrf_unsupported_xr?
-    end
+    # def fail_if_vrf_unsupported_xr(reason)
+    #   fail Cisco::UnsupportedError.new(
+    #     'bgp_af', reason, 'set',
+    #     "#{reason} is not configurable on a " \
+    #     'per-VRF basis on IOS XR') if vrf_unsupported_xr?
```

Comment:
Driving towards metafunctions; the vrf unsupported exceptions are more compactly represented in a table rather than individual functions with individual fail calls.  This will also allow merging the test matrix with the base code - previously the tests replicated the base code.  Now that I've drawn most of the data into the tests, the structure of the command tree can be queried and the appropriate result generated.  Right now, that means throwing a CliError for almost everything, but that would be easy to change.